### PR TITLE
Fix wrong metadata for 1.11

### PR DESCRIPTION
**Pull request by RotBolt**
cc @harshmadhani  @gsmet

### CODE REVIEW

**Review by harshmadhani — APPROVED** [review]
LGTM

File: descriptor/src/main/resources/extensions-support-overrides.json [hunk 1]
```diff
@@ -156,14 +156,6 @@
...
        }
     },
-    {
-      "group-id": "io.quarkus",
-      "artifact-id":"quarkus-jdbc-h2",
-      "metadata" : {
-        "redhat-support" : [
-          "dev-support" ]
-      }
-    },
```
Comment (gsmet) [inline descriptor/src/main/resources/extensions-support-overrides.json:166]: Why did you remove H2?

Also the original issue is talking about Derby being supported but I don't see it anywhere in the file?

Comment (gsmet) [reply descriptor/src/main/resources/extensions-support-overrides.json:166]: @rotbolt ?

Comment (tqvarnst) [reply descriptor/src/main/resources/extensions-support-overrides.json:166]: @RotBolt The question here is if we are productizing the h2 extensions (e.g. based on productized artefact). Then it should be marked as Dev supported. I know the tin the sheet we are not listing dev supported, but it depends on if this extension is productized. If it is we have to mark it as Dev supported to make sure that it's not used in production by customers.

Comment (RotBolt) [reply descriptor/src/main/resources/extensions-support-overrides.json:166]: Ok I will add back the H2, in the sheet it was not mentioned as Dev support so I removed it.
@gsmet . for the derby issue, derby is actually unsupported as per sheet shared with us
Apologize with late response. I did not get any github notification for this

Comment (tqvarnst) [reply descriptor/src/main/resources/extensions-support-overrides.json:166]: I've confirmed with Harsh and @RotBolt that H2 is not productized and therefor should not be marked as dev supported either, so removing them is correct. I believe that the same is true for Derby.
